src/Omnipay/Common/Message/AbstractResponse.php

```diff
@@ -148,6 +148,16 @@ public function getTransactionReference()
     }
 
     /**
+     * Get the transaction ID as generated by the merchant website.
+     *
+     * @return string
+     */
+    public function getTransactionId()
```

Comment:
Not all responses will be able to return a `transactionId`, just like for the `transactionReference`, so I don't think it should be an abstract method. I can't think why it is there at all, other than to plant a flag as to what gateways should be using where they need to.

If it were an abstract method, it should not be too difficult to roll out, since most gateways extend the `AbstractResponse` with their own, so necessary changes would be limited to that.

With the `transactionId`, most code with a response message will already have the transaction ID, so will not be asking the response message for it. Where is is useful, is in the notify handlers, where they are call out of the user's session, and so need to get it from the response message. Perhaps that means just putting the abstract method into a `AbstractNotifyResponse`